diff --git vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php
index b3fb7e0..5880194 100644
--- vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php
+++ vendor/cboden/ratchet/src/Ratchet/Server/IoServer.php
@@ -54,10 +54,14 @@ class IoServer {
      * @param  string                             $address    The address to receive sockets on (0.0.0.0 means receive connections from any)
      * @return IoServer
      */
-    public static function factory(MessageComponentInterface $component, $port = 80, $address = '0.0.0.0') {
+    public static function factory(MessageComponentInterface $component, $port = 80, $address = '0.0.0.0', $sslconf = null) {
         $loop   = LoopFactory::create();
         $socket = new Reactor($address . ':' . $port, $loop);
 
+        if (is_array($sslconf)) {
+            $socket = new SecureReactor($socket, $loop, $sslconf);
+        }
+
         return new static($component, $socket, $loop);
     }
 
